src/main/kotlin/net/maizegenetics/phgv2/pathing/BuildKmerIndex.kt

```diff
@@ -517,12 +489,66 @@ class BuildKmerIndex: CliktCommand(help="Create a kmer index for a HaplotypeGrap
             }
         }
 
-        fun getBufferedWriter(filePath: String): BufferedWriter {
-            return if (filePath.endsWith(".gz")) {
-                GZIPOutputStream(FileOutputStream(filePath)).bufferedWriter()
-            } else {
-                Files.newBufferedWriter(Paths.get(filePath))
+        /**
+         * sampleContigList is a list of Strings of the form contig@sample. otherRegionsList is a list
+         * of Strings of the form contig@sample:xxx-yyy.
+         */
+        data class AgcLists(val sampleContigList: List<String>, val otherRegionsList: List<String>)
+
+        /**
+         * @param graph a [HaplotypeGraph]
+         * @param contigRangesMap   a map of contigs -> list of [ReferenceRange] for that contig generated by graph.rangesByContig()
+         * @param chr a chromosome (contig)
+         *
+         * @return  an [AgcLists] containing lists that can be used as part of an agc command
+         *
+         * This function uses the alt header regions to make lists of contigs or ranges that can be used as part of
+         * an agc command to get all the sequence for a chromosome from a [HaplotypeGraph].
+         */
+        fun rangeListsForAgcCommand(graph: HaplotypeGraph, contigRangesMap: Map<String, List<ReferenceRange>>, chr: String): AgcLists {
+            //If at least half the ranges in a chromosome come from the sample contig, add the contig to the list of
+            //  contig@sample to be retrieved. Otherwise add the individual ranges to otherRegions.
+            val minRangeCount = 0.5 * contigRangesMap[chr]!!.size
```

Comment:
I would recommend putting a require here making sure that chr is in fact in contigRangesMap.  The way you have it set up it will never be null, but if someone else uses this function and doesn't loop through outside this method in a similar manner it will fail.

Reply:
Done.